common/rfb/CConnection.h

```diff
@@ -242,6 +242,9 @@ namespace rfb {
     bool supportsCursorPosition;
     bool supportsDesktopResize;
     bool supportsLEDState;
+#ifdef HAVE_H264
+    bool useH264Encoding;
+#endif
```

Comment:
Can't we handle this via the existing "preferred encoding" mechanism?

Reply:
Sorry. Handle what? #ifdef is used here to make it possible to not compile this functionality at all. Selection of H264Encoding is done by using "preferred encoding" already. Slightly different disposition of this encoding is fo User Interface: you are able to forbid using H264, even if it supported.

Reply:
This issue still remains.

Reply:
I've commented this one below. This is not an issue. It's not about "preferred encoding" mechanism at all. It's about compiled-in or not support of the whole H264 feature. As it is optional compile-time feature and in some systems it is even disabled by default we need these #ifdefs